src/tqec/templates/scale.py

```diff
@@ -0,0 +1,96 @@
+import marshal
+import typing as ty
+from operator import xor
+
+from tqec.exceptions import TQECException
+
+
+def default_scaling_function(k: int) -> int:
+    return 2 * k
+
+
+def get_fixed_scaling_function(fixed_scale: int) -> ty.Callable[[int], int]:
+    return lambda _: fixed_scale
+
+
+class Dimension:
+    def __init__(
+        self,
+        initial_value: int,
```

Comment:
It seems useless providing an initial value that cannot be reached from the `scale_to`. I think it makes more sense to set the `initial_scale` instead of `initial_value`, just like what we did for templates. When the `is_fixed=True`, it can implicitly set the `scaling_function=lambda _: initial_scale` and unify the logic afterwards.

Reply:
I agree. Removing the setter for the value makes that change safe. It removes the possibility to check the correctness of the input value (is it odd or even?) in the constructors of some classes, but this is a small price to pay, and we can find other ways to ensure that.